packages/go_router/lib/src/delegate.dart

```diff
@@ -167,6 +168,11 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
     }());
   }
 
+  /// The top [GoRouterState], the state of the route that was
+  /// last used in either [GoRouter.go] or [GoRouter.push].
```

Comment:
the doc should mention doing GoRouter.of(context).state will not cause rebuild if the state has changed, consider using GoRouterState.of(context) instead.

This property is useful when you want to access the current state outside Element tree.

Reply:
I'll add it here for the time being, but this does not seems to be the right place for this comment to me. This is not inherent to the state property, it's the same for other properties of the go router if I'm not mistaking ?